xarray/tests/test_weighted.py

```diff
@@ -29,6 +29,22 @@ def test_weighted_weights_nan_raises(as_dataset, weights):
         data.weighted(DataArray(weights))
 
 
+@pytest.mark.parametrize("as_dataset", (True, False))
+@pytest.mark.parametrize("weights", ([np.nan, 2], [np.nan, np.nan]))
+def test_weighted_weights_nan_raises_dask(as_dataset, weights):
+
+    data = DataArray([1, 2]).chunk({"dim_0": -1})
+    if as_dataset:
+        data = data.to_dataset(name="data")
+
+    weights = DataArray(weights).chunk({"dim_0": -1})
+
+    weighted = data.weighted(weights)
```

Comment:
You test that dask does not comoute: 

https://github.com/pydata/xarray/blob/83884a1c6dac4b5f6309dfea530414facc100bc8/xarray/tests/test_dask.py#L189-L190